src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        arr.push({ ...newState });
+        break;
+
+      case 'clear':
+        for (const keys in newState) {
+          delete newState[keys];
+        }
+        arr.push({ ...newState });
+        break;
+
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        arr.push({ ...newState });
```

Comment:
I meant that you should push into the array just once. No need in writing it in all cases.